tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -0,0 +1,216 @@
+
+		$this->mockGetImageSize($location, $dimensions);
+
+		$this->logger->log('debug', 'Test Image Path: '.$location, array('app' => 'MediaMetadata'));
+		$this->logger->log('debug', 'Test Image Height: '.$imageHeight, array('app' => 'MediaMetadata'));
+		$this->logger->log('debug', 'Test Image Width: '.$imageWidth, array('app' => 'MediaMetadata'));
+
+		$this->imageDimension->expects($this->once())
+			->method('setImageId')
+			->with($jpgFile->getId());
+		$this->imageDimension->expects($this->once())
+			->method('setImageHeight')
+			->with($imageHeight);
+		$this->imageDimension->expects($this->once())
+			->method('setImageWidth')
+			->with($imageWidth);
+
+		$this->mapper->expects($this->once())
+			->method('insert')
+			->with($this->imageDimension);
+
+		$this->imagehooks->post_create($jpgFile);
+	}
+
+	/**
+	 * @return object|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	protected function mockFile($fileId) {
+		$file = $this->getMockBuilder('OCP\Files\File')
+			->disableOriginalConstructor()
+			->getMock();
+
+		$file->method('getId')
+			->willReturn($fileId);
+
+		return $file;
+	}
+
+	/**
+	 * @param $path
+	 * @return object|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	protected function mockJpgFile($path, $fileId) {
+		$file = $this->mockFile($fileId);
+
+		$this->mockJpgFileMethods($file, $path);
+
+		return $file;
+	}
+
+	/**
+	 * @param $node
+	 * @param $path
+	 */
+	protected function mockJpgFileMethods($node, $path) {
+		$node->method('getPath')
+			 ->willReturn($path);
+	}
+
+	/**
+	 * @param $location
+	 * @param $dimensions
+	 */
+	private function mockGetImageSize($location, $dimensions) {
+		$this->imagehooks->expects($this->once())
+			->method('getimagesize')
+			->with($location)
+			->willReturn($dimensions);
+	}
+
+	/**
+	 * @return object|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockImageDimensionMapper() {
+		/*return new ImageDimensionMapper(
+			$this->db
+		);*/
+
+		return $this->getMockBuilder('OCA\MediaMetadata\Services\ImageDimensionMapper')
+			->setConstructorArgs($this->db)
+			->getMock();
+
+		/*
+		 * Another Method
+		 * $app = new \OCA\MediaMetadata\AppInfo\Application();
+		 * $container = $app->getContainer();
+		 *
+		 * return new ImageDimensionMapper(
+		 *     $container->query('ServerContainer')->getDb()
+		 * );
+		 */
+	}
+
+	/**
+	 * @return object|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockImageDimension() {
+		//return new ImageDimension();
+
+		return $this->getMockBuilder('OCA\MediaMetadata\Services\ImageDimension')
+			->disableOriginalConstructor()
+			->getMock();
+	}
+
+	/**
+	 * @param array $mockedMethods
+	 * @return object|ImageHooks|\PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockPostCreate(array $mockedMethods = []) {
+		$app = new \OCA\MediaMetadata\AppInfo\Application();
+		$container = $app->getContainer();
+
+		return $this->getMockBuilder('OCA\MediaMetadata\Hooks\ImageHooks')
```

Comment:
I think that instead of recreating the mock object, you should just add the `postCreate` method to `$this->imagehooks`. Not important for this PR since we just want tests for testPostCreate